### ISSUE TITLE
Debugger skips some lines in C# web project on localhost

### ISSUE THREAD

**Issue by avifatal** [dotnet/roslyn#19343]
**Version Used**: 
2017 Preview 2 Version 15.2 26430.1-Preview.
This also happened to me in the 2017 community regular edition,
**Steps to Reproduce**:
Have no idea, very random, In order to fix this I need to rename some variables in the context.

**Actual Behavior**:
See in the screen record below, the debugger I think remember one of the previus code states and debug with the old lines.
https://www.screencast.com/t/UGgql6nw

**Comment by tmat** [comment]
Do you remember if you edited the code while debugging? Perhaps added or deleted a line somewhere above the code that you stepped thru?

**Comment by avifatal** [comment]
Yes I did, it was under edit and continue. But yet again, it is not normal behavior.

**Comment by avifatal** [comment]
Here is a better video: https://www.screencast.com/t/jDyL4K3JUxYD

**Comment by avifatal** [comment]
@tmat , Can you please give me some info how to avoid this? I think most of my bugs are comming from code is not updated when running.
In some cases im running old code and I dont know whats going on....
Thanks

**Comment by tmat** [comment]
I'm looking into the issue. It's likely related to lambdas. I'm not sure if there is anything you can do to avoid it short of not editing the lambda.

**Comment by tmat** [comment]
Could you please share the full source code of the method where you make the edit?

**Comment by tmat** [comment]
Never mind I think I figured out a repro:

```
using System;

class Program
{
    static void Main(string[] args)
    {
        var f1 = new Action<int[], int>((a, _) =>
        {
            foreach (var x in a)
            {
                Console.WriteLine(1); // change to 10 and then to 100
            }
        });

        var f2 = new Action<int[], int>((a, _) =>
        {
            foreach (var x in a)
            {
                Console.WriteLine(20); // place breakpoint here
            }
        });

        f1(new[] { 1, 2 }, 1);
        f2(new[] { 1, 2 }, 1);
    }
}
```

1) Place breakpoint as indicated and F5
2) When the breakpoint is hit change constant ```1``` to ```10``` where indicated
3) Step (F10) twice to get the active statement on ```in``` keyword
4) Change ```10``` to ```100```.

**Comment by avifatal** [comment]
@tmat  yes, that's my case.
Please note that the problem may be very deep in the stack. 
Some times you forget that you are inside a lambda execution, or you even don't know that roslyn is not updated and you have major bugs in the code, in this case (after a full day of debugging), I have see all the changes in GIT and start rename variables till I fix the problem.
 
Is this going to be in the next release?
Thanks!

**Comment by brunofvalli** [comment]
You might be in "Release" as your "Active solution configuration" on Visual Studio. Switch back to "Debug" for the system to behave as it used to.

![image](https://user-images.githubusercontent.com/5949285/42393372-d4e0e31c-811b-11e8-9987-9e654e053c09.png)

**Comment by sadranyi** [comment]
You should make sure you running in Debug mode. if you have multiple projects in the solution ensure you set them to debug and build. To do this right click on the Solution, Click Properties and in the solution properties dialog, click configuration Properties in the left pane. Ensure the Debug configuration is selected and Build checkbox checked. You should be good to go.

![01](https://user-images.githubusercontent.com/564700/44811345-49b0e880-ab89-11e8-96e7-057d3aa9fd07.png)
![02](https://user-images.githubusercontent.com/564700/44811349-4e759c80-ab89-11e8-9260-5fa10dc10ed4.png)
![03](https://user-images.githubusercontent.com/564700/44811353-51708d00-ab89-11e8-881a-ae1d1ec03b8a.png)

**Comment by CyrusNajmabadi** [comment]
Closing out as the repro scenario now works properly when i try it in latest vs.
